### ISSUE TITLE
Texture atlas fails to show new region on Android

### ISSUE THREAD

**Issue by PelegMedia** [godotengine/godot#25660]
**Godot version:**
v3.1.beta3.official

**OS/device including version:**
Android nexus tablet

**Issue description:**
I have a sprite with atlas texture, including several images in a single file.
Setting the atlas to show the first image (first region) works fine.
Later on when the game logic changes the region, to show another image the old one remains.
It works on PC but fails on my android device.
I used a workaround of hiding and showing the sprite.

Code example:

```
$Sprite.texture.region = Rect2((miType + iPowerUp) * BLOCK_WIDTH, 0, BLOCK_WIDTH, BLOCK_HEIGHT)
# Workaround
$Sprite.hide()
$Sprite.show()
```

**Steps to reproduce:**

**Minimal reproduction project:**

**Comment by Chaosus** [comment]
Did it fails with GLES2 on PC ?

**Comment by PelegMedia** [comment]
It works on PC using both GLES2 and GLES3 (without the workaround)

**Comment by PelegMedia** [comment]
By the way, the texture size is 2624 X 64 and each region is 64X64.

**Comment by MisterDave** [comment]
Try setting the texture to 2048 or 4096 width, GPU textures sometime need to be power of 2 on mobile devices.

**Comment by akien-mga** [comment]
Fixed by 74d0ed22.

I would recommend that you use textures which have dimensions as powers of 2 indeed, as mentioned by @MisterDave. But the above-mentioned commit should make your current setup work, even if less efficient.
